src/importexport/midi/internal/midishared/midievent.h

```diff
@@ -38,8 +38,8 @@ class MidiEvent : public engraving::MidiCoreEvent
     MidiEvent(uchar t, uchar c, uchar a, uchar b)
         : MidiCoreEvent(t, c, a, b), _edata(0), _len(0) {}
 
-    const uchar* edata() const { return _edata; }
-    void setEData(uchar* d) { _edata = d; }
+    const uchar* edata() const { return _edata.data(); }
+    void setEData(std::vector<uchar>& d) { _edata = d; }
```

Comment:
To eliminate the copying that happens in the `=` operator, you could let it take an rvalue reference:
```suggestion
    void setEData(std::vector<uchar>&& d) { _edata = d; }
```
so that move assignment will be performed instead of copy assignment.

Then you will need to call it like `event.setEData(std::move(data))`.

Reply:
With those changes my backport now crashes on MIDI _import__

Reply:
Nah, I'm too stupid, used `data.resize(len - 1)` rather thsan `data.resize(len + 1)`